File: scala/sublist/src/main/scala/Sublist.scala
object Sublist{
  val Superlist: Int = 1
  val Sublist: Int = 2
  val Equal: Int = 3
  val Unequal: Int = 4

  def sublist(l1: List[Any], l2: List[Any]): Int = {
    var result:Int = 0
    if (l1.isEmpty && l2.isEmpty){
        return Equal
    }
    if (!l2.isEmpty && l1.isEmpty){
        return Sublist
    }
    if (!l1.isEmpty && l2.isEmpty){
        return Superlist
    }

    var length1 = l1.length
    var length2 = l2.length

    if(length1 == length2){
        return if(l1.equals(l2)) Equal else Unequal
    }else if(length1 < length2){
      for(a <- 0 to (length2 - length1)){
        if(l1.equals(l2.slice(a, a + length1))){
          return Sublist
        }
      }
    }else if(length2 < length1){
      for(a <- 0 to (length1 - length2)){
        if(l2.equals(l1.slice(a, a + length2))){
          return Superlist
        }
      }
    }
    return Unequal
  }
}

File: clojure/sublist/src/sublist.clj
(ns sublist)

(defn check-equality [list1 list2]
  (let [len (count list1)]
    (if (and (= 0 (count list1)) (= 0 (count list2)))
      :equal
      (loop [i 0]
        (if (>= i len)
          :equal
          (if (not= (nth list1 i) (nth list2 i))
            :unequal
            (recur (inc i))))))))

(defn check-sublist [list1 list2]
  (if (= 0 (count list1))
    :sublist
    (let [d (- (count list2) (count list1)) l (count list1)]
      (loop [i 0]
        (if (> i d)
          :unequal
          (if (= :equal (check-equality list1 (subvec list2 i (+ i l))))
            :sublist
            (recur (inc i))))))
  ))

(defn classify [list1 list2]
  (cond
    (= (count list1) (count list2)) (check-equality list1 list2)
    (< (count list1) (count list2)) (check-sublist list1 list2)
    :else (if (= :sublist (check-sublist list2 list1)) :superlist :unequal)))